### PR TITLE
Add ITypeSymbol.IsReadOnly public API

### LINKED ISSUE #23792: [Question] How can I check if a ITypeSymbol is a readonly struct?
I'm trying to create a Roslyn Analyzer which checks whether an InParameter is defined on a non-readonly valuetype (from C# 7.2). I can't find a way to check if an ITypeSymbol is readonly. 

I tried the GetAttributes() method. I expected to have the IsReadOnlyAttribute but it doesn't. It is empty.

`
private static bool IsReadonly(ITypeSymbol type) {
    return type.GetAttributes().Any(a => a.AttributeClass.GetType().FullName == "System.Runtime.CompilerServices.IsReadOnlyAttribute");
}
`

Any suggestions?

I published my code on: [Gist](https://gist.github.com/sonnemaf/0001b88db198be33d83f346a47ad874f)

### PR DESCRIPTION
**Pull request by RikkiGibson**
Resolves #23792

### CODE REVIEW

File: src/Compilers/Core/Portable/Symbols/ITypeSymbol.cs [hunk 1]
```diff
@@ -111,6 +111,11 @@ public interface ITypeSymbol : INamespaceOrTypeSymbol
         /// has no concept of unmanaged types.
         /// </summary>
         bool IsUnmanagedType { get; }
+
+        /// <summary>
+        /// True if the type is a readonly struct.
```
Comment (jasonmalinowski) [inline src/Compilers/Core/Portable/Symbols/ITypeSymbol.cs]: The assumption being that if we ever had readonly classes we'd just change the comment?

Comment (RikkiGibson) [reply src/Compilers/Core/Portable/Symbols/ITypeSymbol.cs]: That's what I was going for.

Comment (mavasani) [reply src/Compilers/Core/Portable/Symbols/ITypeSymbol.cs]: Do we envision a future scenario where a non-NamedTypeSymbol can be readonly? If not, should this property be exposed on `INamedTypeSymbol`?

Comment (333fred) [reply src/Compilers/Core/Portable/Symbols/ITypeSymbol.cs]: I believe so, you could have a readonly array type.

**Comment by RikkiGibson** [comment]
This is being retargeted to the readonly members feature branch because we're hoping to roll out a few changes to `IsReadOnly` symbol APIs on different interfaces at the same time.

File: src/Compilers/Core/Portable/Symbols/ITypeSymbol.cs (hunk 1 above)
Comment (cston) [inline src/Compilers/Core/Portable/Symbols/ITypeSymbol.cs]: >a readonly struct [](start = 32, length = 17)

Consider changing to `True if the type is readonly.`

Comment (RikkiGibson) [reply src/Compilers/Core/Portable/Symbols/ITypeSymbol.cs]: Done.

**Pull request merged**